src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/RecallCitizenHutMessage.java

```diff
@@ -0,0 +1,83 @@
+package com.minecolonies.coremod.network.messages.server.colony.building;
+
+import com.ldtteam.structurize.util.LanguageHandler;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.entity.citizen.AbstractEntityCitizen;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
+import com.minecolonies.coremod.network.messages.server.AbstractBuildingServerMessage;
+import com.minecolonies.coremod.util.TeleportHelper;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.network.NetworkEvent;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Optional;
+
+/**
+ * Used to handle citizen recalls to their hut.
+ */
+public class RecallCitizenHutMessage extends AbstractBuildingServerMessage<IBuilding>
+{
+    /**
+     * Empty public constructor.
+     */
+    public RecallCitizenHutMessage()
+    {
+        super();
+    }
+
+    /**
+     * Creates a message to recall all citizens to their hut.
+     *
+     * @param building {@link AbstractBuildingView}
+     */
+    public RecallCitizenHutMessage(@NotNull final AbstractBuildingView building)
+    {
+        super(building);
+    }
+
+    @Override
+    protected void onExecute(@NotNull final NetworkEvent.Context ctxIn, final boolean isLogicalServer, @NotNull final IColony colony, @NotNull final IBuilding building)
+    {
+        final BlockPos location = building.getPosition();
+        final World world = colony.getWorld();
+        for (final ICitizenData citizenData : building.getAssignedCitizen())
+        {
+            Optional<AbstractEntityCitizen> optionalEntityCitizen = citizenData.getEntity();
```

Comment:
could we here instead do data.setLastpos(buildingpos) -> existing entity remove(or null) -> data.updateEntityIfNecessary() ?

Then we're making sure to cleanup the references and get a valid entity in any case. E.g. when the data somehow has a no longer valid entities but still keeps that